R/FilterState.R

```diff
@@ -251,16 +251,28 @@ FilterState <- R6::R6Class( # nolint
             private$server_inputs("inputs")
           }
 
-          observeEvent(input$enable,
-            {
-              if (isTRUE(input$enable)) {
-                private$enable()
-              } else {
-                private$disable()
-              }
-            },
-            ignoreInit = TRUE
+          # Disable/enable this filter state in response to switch flip.
+          observeEvent(input$enable, {
```

Comment:
destroy_observers should remove all observers created by the FilterState class. It means that all observers need to be stored in this list.
```suggestion
          private$observers$enable <- observeEvent(input$enable, {
```